src/main/java/mate/academy/springbootwebgreqit/controller/BookController.java

```diff
@@ -0,0 +1,43 @@
+package mate.academy.springbootwebgreqit.controller;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.springbootwebgreqit.dto.BookDto;
+import mate.academy.springbootwebgreqit.dto.CreateBookRequestDto;
+import mate.academy.springbootwebgreqit.dto.UpdateBookRequestDto;
+import mate.academy.springbootwebgreqit.service.BookService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/books")
+public  class BookController {
+    private BookService bookService;
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.findById(id);
+    }
+
+    @PostMapping
+    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+        return bookService.save(requestDto);
+    }
+
+    @PutMapping("/{id}")
+    public BookDto updateBook(@PathVariable Long id, UpdateBookRequestDto updateBookRequestDto) {
+        return bookService.update(id,updateBookRequestDto);
```

Comment:
In this PR you should only add update and delete endpoint


Book Endpoints:

GET: /api/books (Retrieve book catalog) (should be done in the previous PR)
GET: /api/books/{id} (Retrieve book details) (should be done in the previous PR)
POST: /api/books (Create a new book) (should be done in the previous PR)
PUT: /api/books/{id} (Update a specific book)